tests/test-repo.c

```diff
@@ -151,6 +151,57 @@ test_repo_equal (Fixture       *fixture,
   g_assert_false (ostree_repo_equal (closed_repo, closed_repo));
 }
 
+static void
+test_repo_get_min_free_space (Fixture *fixture,
+                              gconstpointer test_data)
+{
+  g_autoptr (GKeyFile) config = NULL;
+  g_autoptr(GError) error = NULL;
+  gint i = 0;
+  g_autoptr(OstreeRepo) repo = ostree_repo_create_at (fixture->tmpdir.fd, ".",
+                                                       OSTREE_REPO_MODE_ARCHIVE,
+                                                       NULL,
+                                                       NULL, &error);
+  g_assert_no_error (error);
+
+  const char *values_to_test[] = { "500MB",
+                                   "0MB",
+                             //    "17179869185GB", /* Overflow parameter: bytes > G_MAXUINT64 */
+                             //      "17179869183GB", /* bytes approx. under G_MAXUINT64 */ //TODO: Bypass failure
```

Comment:
Do both of these values error out in `ostree_repo_prepare_transaction()`? If so, that’s probably the best testing we’re going to get, and there’s no point in coming up with contortions to get more coverage.

Reply:
Yes, they both error out in `ostree_repo_prepare_transaction` with different appropriate error messages.

> If so, that’s probably the best testing we’re going to get, and there’s no point in coming up with > contortions to get more coverage.

Can you explain this a bit more? Does it mean there is no need for `0MB`, `500MB` test etc.?

Reply:
As in, don’t try and make the `prepare_transaction()` call artificially succeed so you can test those large values with the `get_min_free_space_bytes()` call.